backends/candle/src/models/nomic.rs

```diff
@@ -92,48 +95,336 @@ impl NomicBertEmbeddings {
 }
 
 pub struct NomicBertGatedMLP {
-    gate_up_proj: Linear,
-    down_proj: Linear,
+    fc11: Linear,
+    fc12: Linear,
+    fc2: Linear,
+    activation: HiddenAct,
 
     span: tracing::Span,
 }
 
 impl NomicBertGatedMLP {
     pub fn load(vb: VarBuilder, config: &NomicConfig) -> Result<Self> {
         let intermediate_size = config.n_inner;
+        let activation = config.activation_function.clone();
 
-        let gate_proj_weight = vb
-            .pp("fc12")
+        let fc11_weight = vb
+            .pp("fc11")
             .get((intermediate_size, config.n_embd), "weight")?;
+        let fc11_bias = if config.mlp_fc1_bias {
+            Some(vb.pp("fc11").get((intermediate_size,), "bias")?)
+        } else {
+            None
+        };
+        let fc11 = Linear::new(fc11_weight, fc11_bias, None);
 
-        let up_proj_weight = vb
-            .pp("fc11")
+        let fc12_weight = vb
+            .pp("fc12")
             .get((intermediate_size, config.n_embd), "weight")?;
+        let fc12_bias = if config.mlp_fc1_bias {
+            Some(vb.pp("fc12").get((intermediate_size,), "bias")?)
+        } else {
+            None
+        };
+        let fc12 = Linear::new(fc12_weight, fc12_bias, None);
+
+        let fc2_weight = vb
+            .pp("fc2")
+            .get((config.n_embd, intermediate_size), "weight")?;
+        let fc2_bias = if config.mlp_fc2_bias {
+            Some(vb.pp("fc2").get((config.n_embd,), "bias")?)
+        } else {
+            None
+        };
+        let fc2 = Linear::new(fc2_weight, fc2_bias, None);
+
+        Ok(Self {
+            fc11,
+            fc12,
```

Comment:
The fusing of gate_up into a single tensor, is done on purpose as it's faster to do a single matmul, can you fuse them again ?

Reply:
sure! I'll work on it too!

Reply:
I've changed `fc11` and `fc12` into a single linear layer `fc1`.

https://github.com/huggingface/text-embeddings-inference/pull/596/commits/95812261b26b0873277d7755dd97418ac9fa187b